noir-projects/aztec-nr/aztec/src/messages/logs/note.nr

```diff
@@ -1,31 +1,19 @@
 use crate::{
-    context::PrivateContext,
-    event::event_interface::MessageDelivery,
     messages::{
         encoding::encode_message,
         encryption::{aes128::AES128, log_encryption::LogEncryption},
         logs::utils::prefix_with_tag,
-        msg_type::{PARTIAL_NOTE_PRIVATE_MSG_TYPE_ID, PRIVATE_NOTE_MSG_TYPE_ID},
-        offchain_messages::emit_offchain_message,
+        msg_type::PARTIAL_NOTE_PRIVATE_MSG_TYPE_ID,
     },
-    note::{note_emission::NoteEmission, note_interface::NoteType},
-    utils::remove_constraints::remove_constraints_if,
+    note::note_interface::NoteType,
 };
 use protocol_types::{
-    abis::note_hash::NoteHash,
     address::AztecAddress,
     constants::{PRIVATE_LOG_CIPHERTEXT_LEN, PRIVATE_LOG_SIZE_IN_FIELDS},
     traits::Packable,
 };
 
-// TODO: it feels like this existence check is in the wrong place. In fact, why is it needed at all? Under what circumstances have we found a non-existent note being emitted accidentally?
-fn assert_note_exists(context: PrivateContext, note_hash_counter: u32) {
-    // TODO(#8589): use typesystem to skip this check when not needed
-    let note_exists =
-        context.note_hashes.storage().any(|n: NoteHash| n.counter == note_hash_counter);
-    assert(note_exists, "Can only emit a note log for an existing note.");
-}
-
+// TODO(#16881): once partial notes support emission via an off-chain message we will most likely want to remove this.
 pub fn compute_partial_note_log<Note>(
```

Comment:
I expect this to be nuked once partial notes support message emission as an off-chain message as then, in the implementation of the partial note, we will need to handle the 2 delivery modes and that will result in us needing to control when to prefix the message ciphertext with tag and when not.